fastjet/internal/delaunay/triangulation.go

```diff
@@ -0,0 +1,663 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package delaunay
+
+import (
+	"fmt"
+)
+
+// Delaunay holds necessary information for the
+// delaunay triangulation
+type Delaunay struct {
+	// triangles is a slice of all triangles that have been created. It is used to get the final
+	// list of triangles in the delaunay triangulation
+	triangles triangles
+	// root is a triangle that contains all points. It is used as the starting point in the hierarchy
+	// to find the triangle that contains a point
+	root                   *Triangle
+	maxX, minX, maxY, minY float64
+}
+
+// NewDelaunay creates a delaunay triangulation with the given points
+// all points have to be inside the user defined bounds
+func NewDelaunay(points []*Point, maxX float64, minX float64, maxY float64, minY float64) *Delaunay {
```

Comment:
1.  With the Point interface this would take `points []Point.
2. I'm not convinced about the need of providing a bounding rectangle. What if the points are not known a priori? Currently the code panics if the rectangle is wrong. It seems to me that the rectangle exists only for building the tree for quickly locating the containing triangle. How does it compare to other approaches, for example "walking" on the mesh towards to point. This would need no bounding rectangle, and no additional data structure. If giving the bounding rectangle were significantly faster, providing it could be optional.
3. Currently `points` can be nil and providing them has no benefit. With some preprocessing assumptions could be made and the initial points could be inserted more efficiently. Otherwise `NewDelaunay` could be simplified to just:
```
t := delaunay.New()
for _, p := range points {
    t.InsertPoint(p)
}
```
with some mechanism for specifying an implementation of geometric predicates, perhaps the bounding rectangle (if considered useful) and any other settings.
4. The bounds should be given as (min, max).